### DISCUSSION TITLE
Storing client and torrent data separately

### DISCUSSION THREAD

**Discussion by FIGBERT** [anacrolix/torrent#584] [Q&A] [answered]
Heyo!

Is it possible to store client (meta)data (like `.torrent.db` and friends) in a separate directory from torrent downloads? I'd like to throw the client metadata in a temporary folder, and the torrent data in a user-determined location. Additionally, is it possible to set the download location for torrents on a per-torrent basis?

One workaround I've thought of to fix this issue as I currently understand it is to download everything in a temporary directory and then copy it out once the torrent is done, but that feels quite janky.

Thanks!

**Comment by anacrolix** [comment, marked as answer]
For storing the metadata separately, override https://pkg.go.dev/github.com/anacrolix/torrent@v1.31.0/storage#PieceCompletion in your storage configuration.

For modifying storage configuration to handle torrents separately, or to change the default paths etc see:

 * DefaultStorage in https://pkg.go.dev/github.com/anacrolix/torrent#ClientConfig
 * Storage in https://pkg.go.dev/github.com/anacrolix/torrent#TorrentSpec which overrides the above
 * Provided client storage implementations like for files that provide options to configure directories on a per-Torrent basis https://pkg.go.dev/github.com/anacrolix/torrent@v1.31.0/storage#NewFileClientOpts
